File: HubbardAdvanced/dimension/Get_K_3d.m
function K = Get_K_3d(NumInEdge)
    NumOfVertexs = NumInEdge^3;
    K = zeros([NumOfVertexs,NumOfVertexs]);
    for i = 1:1:NumOfVertexs
        for j = 1:1:NumOfVertexs
            [xi,yi,zi] = IndexToCoor_3d(i,NumInEdge);
            [xj,yj,zj] = IndexToCoor_3d(j,NumInEdge);
            delta_x = min(abs(xi-xj),NumInEdge - abs(xi-xj));
            delta_y = min(abs(yi-yj),NumInEdge - abs(yi-yj));
            delta_z = min(abs(zi-zj),NumInEdge - abs(zi-zj));
            if (delta_x + delta_y + delta_z) == 1
                K(i,j) = 1;
            end
        end
    end%Buide Up Matrix K
end

File: Armadillo/Get_K_3d.cpp
#include "DQMC.h"

mat Get_K_3d(int NumInEdge)
{
	int NumOfVertexs = NumInEdge * NumInEdge * NumInEdge;
	mat K(NumOfVertexs,NumOfVertexs, fill::zeros);
	for (int i = 0; i < NumOfVertexs; i++)
	{
		for (int j = 0; j < NumOfVertexs; j++)
		{
			int xi, yi, zi;
			int xj, yj, zj;
			IndexToCoor_3d(&xi, &yi, &zi, i, NumInEdge);
			IndexToCoor_3d(&xj, &yj, &zj, j, NumInEdge);
			int delta_x, delta_y, delta_z;
			delta_x = min(abs(xi - xj), NumInEdge - abs(xi - xj));
			delta_y = min(abs(yi - yj), NumInEdge - abs(yi - yj));
			delta_z = min(abs(zi - zj), NumInEdge - abs(zi - zj));
			if (delta_x + delta_y + delta_z == 1)
			{
				K(i, j) = 1;
			};
		};
	};
	return K;
}